pallets/capacity/src/types.rs

```diff
@@ -93,17 +93,32 @@ impl<T: Config> StakingAccountDetails<T> {
 	}
 
 	/// Decrease the amount of active stake by an amount and createa an UnlockChunk.
-	pub fn decrease_by(&mut self, amount: BalanceOf<T>, thaw_at: T::EpochNumber) -> DispatchResult {
-		let new_active = self.active.saturating_sub(amount);
-
-		let unlock_chunk = UnlockChunk { value: amount, thaw_at };
-
-		self.active = new_active;
+	pub fn deduct(
+		&mut self,
+		amount: BalanceOf<T>,
+		thaw_at: T::EpochNumber,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		// let's check for an early exit before doing all these calcs
```

Comment:
These are very simple calculations right? I guess there is one constant pull, but I wouldn't think the savings could be very much for the math.

Reply:
My thinking was, not individually no, but for a network with millions of txns it can add up.